### PR TITLE
refactor the code

### PR DESCRIPTION
**Pull request by guoweilkd**
1. Add a memory free interface:`lv_ft_font_destory()`
2. fix atuo free FT_face bug when use cache manager. 
3. For different sizes of the same font, use FT_size instead of FT_Face

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,23 +11,32 @@ Interface to FreeType to generate font bitmaps run time
...
 ```c
-lv_freetype_init(64); /*Cache max 64 glyphs*/
+/* init freetype library */
+lv_freetype_init(64, 1, 0);
 
 /*Create a font*/
-static lv_font_t font1;
-lv_freetype_font_init(&font1, "./lv_lib_freetype/arial.ttf", 32);
+lv_ft_info_t info;
+info.name = "./lv_lib_freetype/simhei.ttf";
```
Comment (embeddedt) [inline README.md:23]: Please leave it as Arial, as that is probably the more common font and simhei.ttf is not in the repository.

File: README.md [hunk 2]
```diff
@@ -11,23 +11,32 @@ Interface to FreeType to generate font bitmaps run time
 - Add library: `freetype` (for GCC: `-L/usr/local/lib -lfreetype`)
 
 ## Usage in LVGL
+
+To enable cache, set`LV_USE_FT_CACHE_MANGER  1`in lv_freetype.h.
```
Comment (embeddedt) [inline README.md:15]: This is a nitpick, but if we don't fix typos now, they will be in the API for months or years. Please change to `MANAGER`.

File: README.md [hunk 3]
```diff
@@ -11,23 +11,32 @@ Interface to FreeType to generate font bitmaps run time
...
-lv_style_set_text_font(&style, LV_STATE_DEFAULT, &font1);
+lv_style_set_text_font(&style, LV_STATE_DEFAULT, info.font);
 
 /*Create a label with the new style*/
 lv_obj_t * label = lv_label_create(lv_scr_act(), NULL);
 lv_obj_add_style(label, LV_LABEL_PART_MAIN, &style);
 lv_label_set_text(label, "Hello world");
 
+/* free font */
+//lv_ft_font_destory(info.font);
```
Comment (embeddedt) [inline README.md:39]: Please rename.

```suggestion
//lv_ft_font_destroy(info.font);
```

File: lv_freetype.c [hunk 4]
```diff
@@ -63,41 +262,41 @@ static bool get_glyph_dsc_cache_cb(const lv_font_t * font, lv_font_glyph_dsc_t *
...
+    desc_sbit_type.face_id = (FTC_FaceID)face;
+    desc_sbit_type.flags = FT_LOAD_RENDER | FT_LOAD_TARGET_NORMAL;
+    desc_sbit_type.height = dsc->weight;
+    desc_sbit_type.width = dsc->weight;
+    FT_UInt charmap_index = FT_Get_Charmap_Index(face->charmap);
+    FT_UInt glyph_index = FTC_CMapCache_Lookup(cmap_cache, (FTC_FaceID)face, charmap_index, unicode_letter);
+    FT_Error error = FTC_SBitCache_Lookup(sbit_cache, &desc_sbit_type, glyph_index, &sbit, NULL);
 
     dsc_out->adv_w = sbit->xadvance;
-    dsc_out->box_h = sbit->height;         /*Height of the bitmap in [px]*/
```
Comment (embeddedt) [inline lv_freetype.c:84]: The comments will probably be useful for future readers... any reason why they were removed?

**Review by embeddedt — COMMENTED** [review]
A few general comments.

**Comment by kisvegabor** [comment]
Thank you!
It's great to expose all caching parameters to `lv_freetype_init`. I think in most cases `max_bytes` is the easiest to use.

File: README.md [hunk 5]
```diff
@@ -11,23 +11,32 @@ Interface to FreeType to generate font bitmaps run time
 - Add library: `freetype` (for GCC: `-L/usr/local/lib -lfreetype`)
 
 ## Usage in LVGL
+
+To enable cache, set`LV_USE_FT_CACHE_MANAGER  1`in lv_freetype.h.
+
 ```c
-lv_freetype_init(64); /*Cache max 64 glyphs*/
+/* init freetype library */
+lv_freetype_init(64, 1, 0);
```
Comment (kisvegabor) [inline README.md:19]: Could you add a little bit more comment here about what 64, 1, 0 means?

Comment (guoweilkd) [reply README.md:19]: Interface specification added to lv_freetype.h

**Comment by kisvegabor** [comment]
Looks good! Thank you very much!

**Pull request merged**